Guide/validation.markdown

```diff
@@ -213,6 +213,46 @@ Then just call it like:
 user |> validateField #age isAge`
 ```
 
+### Creating a custom validator that uses IO
+
+Modify the `buildPost` function like:
+
+```haskell
+-- Add type signature specifying that it requires the Model Context and Controller Context
+-- to be available and that the function’s return value will be IO Post rather than Post.
+buildPost :: (?modelContext :: ModelContext, ?context :: ControllerContext) => Post -> IO Post
+buildPost post = post
+    |> fill @'["title", "body"]
+    -- Add custom validator. The below example checks if the post’s title is unique.
+    |> (\post -> validateFieldIO #title (titleIsUnique post) post) 
```

Comment:
```suggestion
    |> validateFieldIO #title (titleIsUnique post)
```

Reply:
Is `post`, at this point that one that was piped and filled?

Reply:
oh, true, i missed that, sorry. Then it needs to stay how it is right now 👍